cli/flox/src/config/mod.rs

```diff
@@ -73,6 +73,11 @@ pub struct FloxConfig {
     /// The URL of the FloxHub instance to use
     pub floxhub_url: Option<Url>,
 
+    /// The URL of the catalog instance to use
+    // Using a URL here adds an extra trailing slash,
+    // so just use a String.
```

Comment:
I think at this point it should be global. And it causes an error because the generated client code expects something without a trailing slash as baseurl:
```
> git diff --no-ext-diff
diff --git a/cli/flox/src/config/mod.rs b/cli/flox/src/config/mod.rs
index 27e01ea9..a731a478 100644
--- a/cli/flox/src/config/mod.rs
+++ b/cli/flox/src/config/mod.rs
@@ -76,7 +76,7 @@ pub struct FloxConfig {
     /// The URL of the catalog instance to use
     // Using a URL here adds an extra trailing slash,
     // so just use a String.
-    pub catalog_url: Option<String>,
+    pub catalog_url: Option<Url>,

     /// Rule whether to change the shell prompt in activated environments
     pub shell_prompt: Option<EnvironmentPromptConfig>,
diff --git a/cli/flox/src/utils/init/catalog_client.rs b/cli/flox/src/utils/init/catalog_client.rs
index b9825ffa..7720530c 100644
--- a/cli/flox/src/utils/init/catalog_client.rs
+++ b/cli/flox/src/utils/init/catalog_client.rs
@@ -38,7 +38,9 @@ pub fn init_catalog_client(config: &Config) -> Result<Option<Client>, anyhow::Er
         Ok(Some(Client::Mock(MockClient::new(Some(path))?)))
     } else if let Some(ref catalog_url) = config.flox.catalog_url {
         debug!("using catalog client with url: {}", catalog_url);
-        Ok(Some(Client::Catalog(CatalogClient::new(catalog_url))))
+        Ok(Some(Client::Catalog(CatalogClient::new(
+            catalog_url.as_str(),
+        ))))
     } else {
         debug!("using production catalog client");
         Ok(Some(Client::Catalog(CatalogClient::default())))
> FLOX_CATALOG_URL=https://flox-catalog.preview.flox.dev FLOX_FEATURES_USE_CATALOG=true flox show hello
❌ ERROR: unexpected catalog connection error: Invalid Response Payload (b"{\"detail\":\"Not Found\"}"): missing field `status_code` at line 1 column 22: missing field `status_code` at line 1 column 22
```

Reply:
well, that's annoying
1) our server _should_ handle "https://flox-catalog.preview.flox.dev//search" (or what the url is)
2) progenitor _should_ make use of `url` imo.

we can call `catalog_url.as_str().trim_right_matches("/")`. At least we get _some_ benefit of using a url (though urls with query params attached will still make it fall over but we're not handling that for the `floxhub_url` either afaik

Reply:
I don't think we should block on using a type that doesn't serialize the way we need it to. This isn't really user facing and I'm not too worried about needing `Url` validation

Reply:
hrm yea its not critical
i'm not a fan of weakening this necessarily but we better focus on something else for the time being, the config field is at least not very relevant in production cases for a while.